src/Core/src/Layouts/GridLayoutManager.cs

```diff
@@ -990,17 +988,44 @@ void DetermineCellMeasureHeight(Cell cell)
 				{
 					UpdateKnownMeasureHeight(cell);
 				}
-				else if (cell.IsRowSpanAuto)
+				else if (TreatCellHeightAsAuto(cell))
 				{
 					cell.MeasureHeight = double.PositiveInfinity;
 				}
-				else if (cell.IsRowSpanStar && double.IsInfinity(_gridHeightConstraint))
+
+				// For all other situations, we'll have to wait until we've measured the Auto rows
+			}
+
+			bool TreatCellWidthAsAuto(Cell cell)
+			{
+				if (cell.IsColumnSpanAuto)
 				{
-					// Because the Grid isn't vertically constrained, we treat * rows as Auto 
-					cell.MeasureHeight = double.PositiveInfinity;
+					return true;
 				}
 
-				// For all other situations, we'll have to wait until we've measured the Auto rows
+				if (double.IsInfinity(_gridWidthConstraint) && cell.IsColumnSpanStar)
+				{
+					// Because the Grid isn't horizontally constrained, we treat * columns as Auto 
+					return true;
+				}
+
+				return false;
```

Comment:
Can we just return? no need for the if
```suggestion
				// Because the Grid isn't horizontally constrained, we treat * columns as Auto 
				return double.IsInfinity(_gridWidthConstraint) && cell.IsColumnSpanStar;
```